src/vision.cs

```diff
@@ -1775,5 +1916,393 @@ interface VNRequestRevisionProviding {
+
+		[Export ("revision")]
+		VNGenerateObjectnessBasedSaliencyImageRequestRevision Revision { get; set; }
+
+		[Static]
+		[Export ("supportedRevisions", ArgumentSemantic.Copy)]
+		NSIndexSet WeakSupportedRevisions { get; }
+
+		[Static]
+		[Wrap ("GetSupportedVersions<VNGenerateObjectnessBasedSaliencyImageRequestRevision> (WeakSupportedRevisions)")]
+		VNGenerateObjectnessBasedSaliencyImageRequestRevision [] SupportedRevisions { get; }
+
+		[Static]
+		[Export ("defaultRevision")]
+		VNGenerateObjectnessBasedSaliencyImageRequestRevision DefaultRevision { get; }
+
+		[Static]
+		[Export ("currentRevision")]
+		VNGenerateObjectnessBasedSaliencyImageRequestRevision CurrentRevision { get; }
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface VNRecognizedText : NSCopying, NSSecureCoding {
+
+		[Export ("string")]
+		string String { get; }
+
+		[Export ("confidence")]
+		float Confidence { get; }
+
+		[Export ("boundingBoxForRange:error:")]
+		[return: NullAllowed]
+		VNRectangleObservation GetBoundingBox (NSRange range, [NullAllowed] out NSError error);
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (VNRectangleObservation))]
+	[DisableDefaultCtor]
+	interface VNRecognizedTextObservation {
+
+		[Export ("topCandidates:")]
+		VNRecognizedText [] TopCandidates (nuint maxCandidateCount);
+
+		[Static]
+		[Export ("observationWithBoundingBox:")]
+		VNRecognizedTextObservation Create (CGRect boundingBox);
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (VNPixelBufferObservation))]
+	interface VNSaliencyImageObservation {
+
+		[NullAllowed, Export ("salientObjects")]
+		VNRectangleObservation [] SalientObjects { get; }
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (VNObservation))]
+	[DisableDefaultCtor]
+	interface VNFeaturePrintObservation {
+
+		[Export ("elementType", ArgumentSemantic.Assign)]
+		VNElementType ElementType { get; }
+
+		[Export ("elementCount")]
+		nuint ElementCount { get; }
+
+		[Export ("data", ArgumentSemantic.Strong)]
+		NSData Data { get; }
+
+		[Internal]
+		[Export ("computeDistance:toFeaturePrintObservation:error:")]
+		bool _ComputeDistance (IntPtr outDistance, VNFeaturePrintObservation featurePrint, [NullAllowed] out NSError error);
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	enum VNAnimalIdentifier {
+
+		[DefaultEnumValue]
+		[Field (null)]
+		Unknown = -1,
+
+		[Field ("VNAnimalIdentifierDog")]
+		Dog,
+
+		[Field ("VNAnimalIdentifierCat")]
+		Cat,
+	}
+
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof (VNImageBasedRequest))]
+	[DisableDefaultCtor]
+	interface VNRecognizeAnimalsRequest {
+
+		[Static]
+		[Export ("knownAnimalIdentifiersForRevision:error:")]
+		[return: NullAllowed]
+		[return: BindAs (typeof (VNAnimalIdentifier []))]
+		NSString [] GetKnownAnimalIdentifiers (VNRecognizeAnimalsRequestRevision requestRevision, [NullAllowed] out NSError error);
```

Comment:
I think `requestRevision` should just be `revision`. It's the **revision** for `VNRecognizeAnimalsRequest`, a `VNRecognizeAnimalsRequestRevision` object.